iothub/device/src/Edge/CustomCertificateValidator.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+namespace Microsoft.Azure.Devices.Client.Edge
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Linq;
+    using System.Net.Security;
+    using System.Security.Cryptography.X509Certificates;
+    using Extensions;
+    using Transport.Mqtt;
+
+    internal class CustomCertificateValidator : ICertificateValidator
+    {
+        private readonly IEnumerable<X509Certificate2> certs;
+        private readonly ITransportSettings[] transportSettings;
+
+        public CustomCertificateValidator(IList<X509Certificate2> certs, ITransportSettings[] transportSettings)
+        {
+            this.certs = certs;
+            this.transportSettings = transportSettings;
+        }
+
+        public void SetupCertificateValidation()
+        {
+            Debug.WriteLine("CustomCertificateValidator.SetupCertificateValidation()");
+
+            var certValidationFunc = this.GetCustomCertificateValidation();
```

Comment:
This API looks a bit odd. SetupCertificateValidation is calling GetCustomCertificateValidation...?

Reply:
Added Create method that does the setup

---
In reply to: [193910624](https://github.com/Azure/azure-iot-sdk-csharp/pull/501#discussion_r193910624) [](ancestors = 193910624)